cloudinit/cmd/main.py

```diff
@@ -386,7 +386,12 @@ def main_init(name, args):
                 LOG.debug("manual cache clean found from marker: %s", mfile)
                 existing = "trust"
 
-        init.purge_cache()
+        # rm_instance_lnk is now set to True. Else, cache will be used in case
+        # instead of optical disk to fetch network config. This check make sure 
+        # that customer always uses latest config data even if the instance-id 
+        # is matching
+        init.purge_cache(True)
```

Comment:
One more comment:

This is a pretty big change to cloud-init. What justification do you have for it? Also, which datasources have you tested this on? Wouldn't this break any user using [nocloud](https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html#method-1-local-filesystem-labeled-filesystem) with only a `CIDATA` drive?